packages/commands/src/solana/deposit.ts

```diff
@@ -0,0 +1,166 @@
+import * as anchor from "@coral-xyz/anchor";
+import { Wallet } from "@coral-xyz/anchor";
+import {
+  AccountLayout,
+  ASSOCIATED_TOKEN_PROGRAM_ID,
+  TOKEN_PROGRAM_ID,
+} from "@solana/spl-token";
+import { clusterApiUrl, PublicKey } from "@solana/web3.js";
+import GATEWAY_DEV_IDL from "@zetachain/protocol-contracts-solana/dev/idl/gateway.json";
+import GATEWAY_PROD_IDL from "@zetachain/protocol-contracts-solana/prod/idl/gateway.json";
+import * as bip39 from "bip39";
+import bs58 from "bs58";
+import { Command, Option } from "commander";
+import { ethers } from "ethers";
+
+const networks = ["devnet", "localnet", "mainnet"];
+
+export interface DepositOptions {
+  amount: string;
+  from: string;
+  mint: string;
+  mnemonic: string;
+  network: string;
+  privateKey: string;
+  recipient: string;
+  to: string;
+  tokenProgram: string;
+}
+
+export const keypairFromMnemonic = async (
+  mnemonic: string
+): Promise<anchor.web3.Keypair> => {
+  const seed = await bip39.mnemonicToSeed(mnemonic);
+  const seedSlice = new Uint8Array(seed).slice(0, 32);
+  return anchor.web3.Keypair.fromSeed(seedSlice);
+};
+
+export const keypairFromPrivateKey = (
+  privateKey: string
+): anchor.web3.Keypair => {
+  try {
+    const decodedKey = bs58.decode(privateKey);
+    return anchor.web3.Keypair.fromSecretKey(decodedKey);
+  } catch (error) {
+    throw new Error(
+      "Invalid private key format. Expected base58-encoded private key."
+    );
+  }
+};
+
+const main = async (options: DepositOptions) => {
+  // Mainnet and devnet use the same IDL
+  const gatewayIDL =
+    options.network === "localnet" ? GATEWAY_DEV_IDL : GATEWAY_PROD_IDL;
+
+  let keypair: anchor.web3.Keypair;
+  if (options.privateKey) {
+    keypair = keypairFromPrivateKey(options.privateKey);
+  } else if (options.mnemonic) {
+    keypair = await keypairFromMnemonic(options.mnemonic);
+  } else {
+    throw new Error("Either privateKey or mnemonic must be provided");
+  }
+
+  let API = "http://localhost:8899";
+  if (options.network === "devnet") {
+    API = clusterApiUrl("devnet");
+  } else if (options.network === "mainnet") {
+    API = clusterApiUrl("mainnet-beta");
+  }
+
+  const connection = new anchor.web3.Connection(API);
+
+  const provider = new anchor.AnchorProvider(
+    connection,
+    new Wallet(keypair),
+    {}
```

Comment:
Do we really need this? I would imagine this will work the same without this empty object.

Reply:
c6d8cca